{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "92b0b4f8",
   "metadata": {},
   "source": [
    "# Libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "c987b99b",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\enhi\\AppData\\Local\\Temp\\ipykernel_6556\\3827688388.py:5: UserWarning: Shapely 2.0 is installed, but because PyGEOS is also installed, GeoPandas will still use PyGEOS by default for now. To force to use and test Shapely 2.0, you have to set the environment variable USE_PYGEOS=0. You can do this before starting the Python process, or in your code before importing geopandas:\n",
      "\n",
      "import os\n",
      "os.environ['USE_PYGEOS'] = '0'\n",
      "import geopandas\n",
      "\n",
      "In a future release, GeoPandas will switch to using Shapely by default. If you are using PyGEOS directly (calling PyGEOS functions on geometries from GeoPandas), this will then stop working and you are encouraged to migrate from PyGEOS to Shapely 2.0 (https://shapely.readthedocs.io/en/latest/migration_pygeos.html).\n",
      "  import geopandas as gpd\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import xarray as xr\n",
    "import rioxarray as rxr\n",
    "from shapely.geometry import mapping\n",
    "import geopandas as gpd\n",
    "from rasterio.enums import Resampling\n",
    "from datetime import datetime, timedelta\n",
    "from calendar import monthrange"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "1d266bd4",
   "metadata": {},
   "outputs": [],
   "source": [
    "main_dir = r'C:\\Users\\enhi\\OneDrive - gmv.com\\Documents\\Projects\\ANIN\\Generating Indices\\FAPAR anomaly\\\\'\n",
    "# Directory to the 8days clipped FAPAR Glass to the AOI\n",
    "FAPAR_Glass = os.path.join(main_dir, 'Clipped_FAPAR')\n",
    "# Directory to an example of Copernicus FAPAR to take the shape and crs fro it\n",
    "Copernicus_ex =rxr.open_rasterio(r'C:\\Users\\enhi\\OneDrive - gmv.com\\Documents\\Projects\\ANIN\\Generating Indices\\FAPAR anomaly\\Monthly_FAPAR_Glass\\April\\2000April.tif', masked =True)\n",
    "# Directory to the AOI\n",
    "Boundary = os.path.join(main_dir, 'Boundary')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "2a000376",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Shapefile loaded. To prepare for masking, run the function\n",
      "        `select_shape`.\n",
      "Mask created.\n"
     ]
    }
   ],
   "source": [
    "# Load the shapefile\n",
    "def load_shape_file(filepath):\n",
    "    \"\"\"Loads the shape file desired to mask a grid.\n",
    "    Args:\n",
    "        filepath: Path to *.shp file\n",
    "    \"\"\"\n",
    "    shpfile = gpd.read_file(filepath)\n",
    "    print(\"\"\"Shapefile loaded. To prepare for masking, run the function\n",
    "        `select_shape`.\"\"\")\n",
    "    return shpfile\n",
    "\n",
    "#Create the mask\n",
    "def select_shape(shpfile):\n",
    "\n",
    "    \"\"\"Select the submask of interest from the shapefile.\n",
    "    Args:\n",
    "        shpfile: (*.shp) loaded through `load_shape_file`\n",
    "        category: (str) header of shape file from which to filter shape.\n",
    "            (Run print(shpfile) to see options)\n",
    "        name: (str) name of shape relative to category.\n",
    "           Returns:\n",
    "        shapely polygon\n",
    "    \"\"\"\n",
    "\n",
    "    col_code = 'ISO3_CODE'\n",
    "    country_codes = ['ZAF', 'LSO', 'SWZ']\n",
    "\n",
    "    # Extract the rows that have 'ZAF', 'LSO', or 'SWZ' in the 'SOV_A3' column\n",
    "    selected_rows = shpfile[shpfile[col_code].isin(country_codes)]\n",
    "\n",
    "    # Combine the selected polygons into a single polygon\n",
    "    unioned_polygon = selected_rows.geometry.unary_union\n",
    "\n",
    "    # Convert the unioned polygon to a geopandas dataframe with a single row\n",
    "    mask_polygon = gpd.GeoDataFrame(geometry=[unioned_polygon])\n",
    "    \n",
    "    print(\"\"\"Mask created.\"\"\")\n",
    "\n",
    "    return mask_polygon\n",
    "shpfile = load_shape_file(Boundary + '\\\\'+ 'CNTR_RG_01M_2020_4326.shp')\n",
    "merged = select_shape(shpfile)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "48a3caf0",
   "metadata": {},
   "outputs": [],
   "source": [
    "# crs of FAPAR copernicus\n",
    "Copernicus_ex_crs = Copernicus_ex.rio.crs\n",
    "# Define a crs to the AOI \n",
    "merged.crs =Copernicus_ex_crs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "cf0f5067",
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "The day of the year of the start of the 8 days is extracted from \n",
    "the name of FAPAR file. \n",
    "There are two functions, one if the start data is in the month and the other\n",
    "if the end data is in the month.\n",
    "'''\n",
    "#Function if the start date of FAPAR is in the month.\n",
    "def weights_start(Start_date):\n",
    "    '''\n",
    "    Inputs:\n",
    "         date in the format (dd-mm-yyyy)\n",
    "    returns:\n",
    "        weight based on the number of observation days overlap the month\n",
    "    '''\n",
    "    #Calculation of the number of days on the month \n",
    "    num_days = monthrange(int(Start_date.strftime('%Y')),\n",
    "                          int(Start_date.strftime('%m')))[1]\n",
    "    # The starting day of the 8days FAPAR in the month\n",
    "    day_of_month = int(Start_date.strftime('%d'))\n",
    "    # Number of days bewteen the starting data and the end of the month\n",
    "    diff = (num_days +1) - day_of_month\n",
    "    #Ofcourse if the Number of days bewteen the starting data and the end of \n",
    "    #the month is greater than 8, that means that all 8 days are in the month\n",
    "    if diff > 8:\n",
    "        day_obs = 8 #day_obs: days of observation FAPAR in the month\n",
    "    #If the difference is less than 8 days that means the days of observation\n",
    "    #in that month equal to the number of days bewteen the starting data and \n",
    "    #the end of the month.\n",
    "    else: \n",
    "        day_obs = diff\n",
    "    # Calculaion of the weight based on the days of observation FAPAR\n",
    "    # in the month\n",
    "    weight = (day_obs / num_days)\n",
    "    \n",
    "    return weight\n",
    "#Function if the end date of FAPAR is in the month.\n",
    "def weights_end(End_date):\n",
    "    '''\n",
    "    Inputs:\n",
    "         date in the format (dd-mm-yyyy)\n",
    "    returns:\n",
    "        weight based on the number of observation days overlap the month\n",
    "    '''\n",
    "    #Calculation of the number of days on the month \n",
    "    num_days = monthrange(int(End_date.strftime('%Y')),\n",
    "                          int(End_date.strftime('%m')))[1]\n",
    "    # The ending day of the 8Days FAPAR in the month\n",
    "    day_of_month = int(End_date.strftime('%d'))\n",
    "    # The days of of observation FAPAR is equal to end date,\n",
    "    # but it will be a further steps\n",
    "    #in the following lines of code\n",
    "    day_obs = day_of_month\n",
    "    \n",
    "    weight = (day_obs / num_days)\n",
    "     # Calculaion of the weight based on the days of observation FAPAR\n",
    "     # in the month\n",
    "    return weight\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "0078035c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "source": [
    "'''\n",
    "The code is built based on that there are years folders which contain all\n",
    "the 8days FAPAR data in that year\n",
    "'''\n",
    "months = ['Jan', 'Feb', 'March', 'April', 'May', 'June', 'July', 'Aug', 'Sep',\n",
    "          'Oct', 'Nov', 'Dec']\n",
    "#list of all years of 8days historical data\n",
    "years = sorted(os.listdir(FAPAR_Glass))\n",
    "#Empty list to contain other lists of the monthly FAPAR data for\n",
    "#each month through years\n",
    "#for example, list of monthly FAPAR of January of each year\n",
    "months_through_years = []\n",
    "#looping through 12 months, the looping started by months not years \n",
    "#as the goal is to calculate histrocal max and min values for each month\n",
    "count = 0\n",
    "for month in range(1,13):\n",
    "\n",
    "    #Empty list for the monthly FAOAR data for each month through years\n",
    "    lis_of_m =[]\n",
    "    #looping through years\n",
    "    for year in years:\n",
    "        #Empty list to contain all 8days FAPAR overlapping in the month\n",
    "        month_in_year = []\n",
    "        #looping through the 8days FAPAR of the year\n",
    "        for image in os.listdir(FAPAR_Glass + '\\\\' + year):\n",
    "            #Read the year from name of the file\n",
    "            y = image[16:20]\n",
    "            #Read the day of the year from the name of the file\n",
    "            DOY = image[20:23]\n",
    "            #convert the day of the year to a date in the format (DD-MM-yyyy)\n",
    "            DOY.rjust(3 + len(DOY), '0')\n",
    "            Date_of_start = datetime.strptime(y + \"-\" + DOY, \"%Y-%j\")\n",
    "            Date_of_end = Date_of_start + timedelta(days=8)\n",
    "            #If the starting date is in the month\n",
    "            if int(Date_of_start.strftime('%m')) == month:\n",
    "                #Apply the function to calculate the weight based on \n",
    "                #the starting date\n",
    "                weight = weights_start(Date_of_start)\n",
    "                \n",
    "                #The path of the image\n",
    "                image = FAPAR_Glass + '\\\\' + year + '\\\\' + image\n",
    "                #Open the tiff file of 8days FAPAR\n",
    "                FAPAR = rxr.open_rasterio(image,\n",
    "                                  masked=True)\n",
    "                # getting scale factor\n",
    "                Scale_factor = FAPAR.attrs['scale_factor']\n",
    "                # Calculate real FAPAR values using scale factor\n",
    "                FAPAR =FAPAR * Scale_factor\n",
    "                #Multiply the 8days FAPAR with the weight \n",
    "                img_weighted = weight * FAPAR\n",
    "                #Append th weighted 8days FAPAR to the list of all images \n",
    "                #in that month\n",
    "                month_in_year.append(img_weighted)\n",
    "            #If the ending date is in the month  \n",
    "            elif int(Date_of_end.strftime('%m')) == month:\n",
    "                # If the end day is after 8th of the month,\n",
    "                #that means the image was already considered in starting date,\n",
    "                #so nothing will happen.\n",
    "                if int(Date_of_end.strftime('%d')) >7:\n",
    "                    continue\n",
    "                #If the end day is in January, it should not be considered as\n",
    "                #the first image of a year is always starts in 1st of January \n",
    "                if int(Date_of_end.strftime('%m')) == 1:\n",
    "                    continue\n",
    "                    \n",
    "                #Apply the function to calculate the weight based on\n",
    "                #the ending date\n",
    "                weight = weights_end(Date_of_end)\n",
    "                #The path of the image\n",
    "                image = FAPAR_Glass + '\\\\'+ year + '\\\\' + image\n",
    "                #Open the tiff file of 8days FAPAR\n",
    "                FAPAR = rxr.open_rasterio(image,\n",
    "                                  masked=True)\n",
    "                # getting scale factor\n",
    "                Scale_factor = FAPAR.attrs['scale_factor']\n",
    "                # Calculate real FAPAR values using scale factor\n",
    "                FAPAR =FAPAR * Scale_factor\n",
    "\n",
    "                 #Multiply the 8days FAPAR with the weight\n",
    "                img_weighted = weight * FAPAR\n",
    "                \n",
    "                #Append th weighted 8days FAPAR to the list of all images\n",
    "                #in that month\n",
    "                month_in_year.append(img_weighted)\n",
    "              \n",
    "        #Calculating the monthly composite by adding\n",
    "        #the weighted averaged 8days FAPAR overlap this month            \n",
    "        monthly_composite = sum(month_in_year)\n",
    "        \n",
    "        '''\n",
    "        the following condition is to deal with years which has months with no 8days FAPAR data\n",
    "        like 2000 which the data starts by the end of February. At this time the monthly_composite\n",
    "        will be integer, so the following line is to skip these values\n",
    "        '''\n",
    "        if type(monthly_composite) != int:\n",
    "            # Reprojection and resampling of FAPAR Glass to match Copernicus FAPAR\n",
    "            # The data resamlpled from 250m to 300m using nearest neighbour\n",
    "            monthly_composite= monthly_composite.rio.reproject(Copernicus_ex.rio.crs,\n",
    "                                        shape = Copernicus_ex.rio.shape,\n",
    "                                        resampling=Resampling.nearest).rio.reproject_match(Copernicus_ex)\n",
    "            \n",
    "            # Export monthly FAPAR Glass\n",
    "            monthly_composite.rio.to_raster(main_dir + '\\\\' + 'Monthly_FAPAR_Glass'+'\\\\' +year + months[count] + '.tif')\n",
    "\n",
    "    count = count + 1\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "rasterio",
   "language": "python",
   "name": "rasterio"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
